launcher/InstanceCreationTask.cpp

```diff
@@ -42,11 +42,11 @@ void InstanceCreationTask::executeTask()
         setStatus(tr("Removing old conflicting files..."));
         qDebug() << "Removing old files";
 
-        for (auto path : m_files_to_remove) {
+        for (const QString& path : m_files_to_remove) {
             if (!QFile::exists(path))
                 continue;
             qDebug() << "Removing" << path;
-            if (!FS::deletePath(path)) {
+            if (!QFile::remove(path)) {
```

Comment:
This leaves the instance in a weird state at least try to delete all files before emittingfailed

Reply:
This is how the code was before though
Should I change it anyway?

Reply:
If the function is deleting stuff it shouldn't do, stopping is probably a good idea.. I guess of course deleting failing doesn't mean a bug necessarily

Reply:
We are at the last step of the creation and I'm pretty sure we can't rollback this one so my suggestion is to delete all we can before saying to user that it fail due to encountering some folder. That way the user will have a possible working instance even if it fails

Reply:
I doubt the instance could become completely broken from the outcome of this but I will implement this and wait for the opinion of otehrs

Reply:
are this and #2905 trying to solve the same thing?
if so I think it's approach is preferable.

Reply:
I also do want to prevent any kind of update to anything in saves even if it does not involve deleting files since it might cause a lot of weirdness if the local copy has been modified. That pr only prevents deletions, though, so I would prefer something on the level of both modrinth and curseforge which handles the extraction.